src/main/java/com/rapid7/appspider/ScanManagement.java

```diff
@@ -108,30 +108,38 @@ public static String getScanStatus(String restUrl, String authToken, String scan
      * @param restUrl
      * @param authToken
      * @param scanId
-     * @return
+     * @return  Boolean on success representing whether the report is present or not on success, or null on failure
      */
-    public static Boolean hasReport(String restUrl, String authToken, String scanId) {
-        String apiCall = restUrl + HASREPORT;
-        Map<String, String> params = new HashMap<String, String>();
-        params.put("scanId", scanId);
-        Object response = get(apiCall, authToken, params);
-        if (response.getClass().equals(JSONObject.class)) {
-            return ((JSONObject)response).getBoolean("Result")&&
-                    ((JSONObject) response).getBoolean("IsSuccess");
-        }
-        return null;
+    public static Optional<Boolean> hasReport(String restUrl, String authToken, String scanId) {
+        return getBooleanResultFromApiCall(restUrl + HASREPORT, authToken, buildScanIdParameters(scanId));
```

Comment:
Being picky but HAS_REPORT is better naming convention for constants

Reply:
updated, as have the other constants to follow the same standard, these were the existing names but your point stands